arches/app/media/js/views/components/cards/photo-gallery-card.js

```diff
@@ -150,31 +155,58 @@ define([
-            var tilevalue = {
-                name: file.name,
-                accepted: true,
-                height: file.height,
-                lastModified: file.lastModified,
-                size: file.size,
-                status: file.status,
-                type: file.type,
-                width: file.width,
-                url: null,
-                file_id: null,
-                index: 0,
-                content: window.URL.createObjectURL(file),
-                error: file.error
-            };
-            newtile.data[self.fileListNodeId]([tilevalue]);
-            newtile.formData.append('file-list_' + self.fileListNodeId, file, file.name);
-            newtile.resourceinstance_id = self.card.resourceinstanceid;
-            if (self.card.tiles().length === 0) {
-                sleep(50);
+            var acceptedFileFormats;
+            var loadFile;
+            acceptedFileFormats = ((ko.unwrap(self.acceptedFiles)).split(',').map(item=>item.trim())).map(format => format.replace('.', ''));
+            if(ko.unwrap(self.acceptedFiles) != "All Formats Accepted" && acceptedFileFormats !== undefined && acceptedFileFormats.length > 0){
+                var fileType = file.name.split('.').pop().toLowerCase();
+                if(acceptedFileFormats.includes(fileType)){
+                    loadFile = true;
+                }
+                else{
+                    loadFile = false;
+                }
+            }
+            else{
+                loadFile = true;
+            }
+
+            if (loadFile === true) {
+                var newtile;
+                newtile = self.card.getNewTile();
+                var tilevalue = {
+                    name: file.name,
+                    accepted: true,
+                    height: file.height,
+                    lastModified: file.lastModified,
+                    size: file.size,
+                    status: file.status,
+                    type: file.type,
+                    width: file.width,
+                    url: null,
+                    file_id: null,
+                    index: 0,
+                    content: window.URL.createObjectURL(file),
+                    error: file.error
+                };
+                newtile.data[self.fileListNodeId]([tilevalue]);
+                newtile.formData.append('file-list_' + self.fileListNodeId, file, file.name);
+                newtile.resourceinstance_id = self.card.resourceinstanceid;
+                if (self.card.tiles().length === 0) {
+                    sleep(50);
+                }
+                newtile.save();
+                self.card.newTile = undefined;
+            }
+            else{
+                params.pageVm.alert(new AlertViewModel(
+                    'ep-alert-red',
+                    "Incorrect File Format",
+                    "The file format is not accepted. Please upload a file in the following formats:" + ko.unwrap(self.acceptedFiles) + ".",
```

Comment:
Could you add translation strings to javascript.htm for these too?

Reply:
@chiatt Added as requested.  I couldn't find a way to use the title/text pattern towards the top of the translations that would allow me to pass in acceptedFiles() so I've put these as two separate translations.  In a way this might be better as it will allow reuse in different scenarios.  I hope that's ok